dali/pipeline/operators/reader/coco_reader_op.h

```diff
@@ -46,7 +46,14 @@ class COCOReader : public DataReader<CPUBackend, ImageLabelWrapper> {
     skip_empty_(spec.GetArgument<bool>("skip_empty")),
     save_img_ids_(spec.GetArgument<bool>("save_img_ids")) {
     ParseAnnotationFiles();
-    loader_.reset(new FileLoader(spec, image_id_pairs_));
+
+    if (spec.HasArgument("file_list"))
+      loader_.reset(new FileLoader(spec));
```

Comment:
For 'file_list` why we are skipping `shuffle_after_epoch`?

Reply:
It was based on the experiments I did - when I used `file_list` I generated list for all the epochs right away. But you are right, it would be better to have this as an option, I will fix it.

Reply:
Done